core/src/hir/lowering.rs

```diff
@@ -989,11 +991,13 @@ impl<'ast> LoweringContext<'ast> {
                             true,
                         );
 
+                        let attrs = self.attr_validator.attr_from_ast(&self_param.attrs, &Attrs::default(), &mut self.errors);
```

Comment:
thought: we should add an `attrs.allowed_on_param_or_field()` method to Attrs and check it here and below. I don't think it's a good idea to let people think they can rename fields/params and then have it not work

(eventually i'd like to support param/field renaming though)